indico/modules/events/contributions/forms.py

```diff
@@ -203,3 +205,22 @@ def validate_name(self, field):
             query = query.filter(ContributionType.id != self.contrib_type.id)
         if query.count():
             raise ValidationError(_("A contribution type with this name already exists"))
+
+
+class ContributionExportTeXForm(IndicoForm):
+    """Form for Tex-based export selection"""
+    format = SelectField(_('Format'), default='PDF')
+    sort_by = IndicoEnumSelectField(_('Sort by'), enum=BOASortField, default=BOASortField.abstract_title,
+                                    sorted=True)
+    contrib_ids = HiddenFieldList()
+    submitted = HiddenField()
+
+    def __init__(self, *args, **kwargs):
+        self.contribs = kwargs.get('contribs')
+        super(ContributionExportTeXForm, self).__init__(*args, **kwargs)
+        self.contrib_ids.data = [c.id for c in self.contribs]
+        self.format.choices = {'PDF': 'PDF',
+                               'TeX': 'Zipped TeX directory'}.items()
```

Comment:
`_('...')` for i18n, and I would just put `[(.., ..), (.., ..)]` here instead of creating a dict and then getting its items

Reply:
Changes with the signal introduction